onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -346,8 +346,10 @@ def list(self, request, *args, **kwargs):
             xform_id, is_merged_dataset = qs[0] if qs else (lookup, False)
             pks = [xform_id]
             if is_merged_dataset:
-                pks = [__ for __ in MergedXForm.objects.values_list('xforms',
-                                                                    flat=True)]
+                pks = [__ for __ in MergedXForm.objects.filter(
```

Comment:
why 2 underscores? is the `if` to filter out nils?

Reply:
A single underscore I might override the `uggettext as _`, the double underscore also keeps pylint silent.
Yes, when a merged dataset has no form linked to it, it returns a nil value here, we need to filter that out.

Reply:
oh right, cool